examples/fungible-token/src/lib.rs

```diff
@@ -0,0 +1,341 @@
+        }
+        // Retrieving the account from the state.
+        let mut account = self.get_account(&owner_id);
+
+        // Checking and updating unlocked balance
+        if account.balance < amount {
+            env::panic(b"Not enough balance");
+        }
+        account.balance -= amount;
+
+        // If transferring by escrow, need to check and update allowance.
+        let escrow_account_id = env::predecessor_account_id();
+        if escrow_account_id != owner_id {
+            let allowance = account.get_allowance(&escrow_account_id);
+            if allowance < amount {
+                env::panic(b"Not enough allowance");
+            }
+            account.set_allowance(&escrow_account_id, allowance - amount);
+        }
+
+        // Saving the account back to the state.
+        self.set_account(&owner_id, &account);
+
+        // Deposit amount to the new owner and save the new account to the state.
+        let mut new_account = self.get_account(&new_owner_id);
+        new_account.balance += amount;
+        self.set_account(&new_owner_id, &new_account);
+    }
+
+    /// Transfer `amount` of tokens from the caller of the contract (`predecessor_id`) to
+    /// `new_owner_id`.
+    /// Act the same was as `transfer_from` with `owner_id` equal to the caller of the contract
+    /// (`predecessor_id`).
+    pub fn transfer(&mut self, new_owner_id: AccountId, amount: U128) {
+        self.transfer_from(env::predecessor_account_id(), new_owner_id, amount);
+    }
+
+    /// Returns total supply of tokens.
+    pub fn get_total_supply(&self) -> U128 {
+        self.total_supply.into()
+    }
+
+    /// Returns balance of the `owner_id` account.
+    pub fn get_balance(&self, owner_id: AccountId) -> U128 {
+        self.get_account(&owner_id).balance.into()
+    }
+
+    /// Returns current allowance of `escrow_account_id` for the account of `owner_id`.
+    ///
+    /// NOTE: Other contracts should not rely on this information, because by the moment a contract
+    /// receives this information, the allowance may already be changed by the owner.
+    /// So this method should only be used on the front-end to see the current allowance.
+    pub fn get_allowance(&self, owner_id: AccountId, escrow_account_id: AccountId) -> U128 {
+        self.get_account(&owner_id).get_allowance(&escrow_account_id).into()
+    }
+}
+
+impl FungibleToken {
+    /// Helper method to get the account details for `owner_id`.
+    fn get_account(&self, owner_id: &AccountId) -> Account {
+        let account_hash = env::sha256(owner_id.as_bytes());
+        self.accounts.get(&account_hash).unwrap_or_else(|| Account::new(account_hash))
+    }
+
+    /// Helper method to set the account details for `owner_id` to the state.
+    fn set_account(&mut self, owner_id: &AccountId, account: &Account) {
+        let account_hash = env::sha256(owner_id.as_bytes());
+        self.accounts.insert(&account_hash, &account);
+    }
+}
+
+metadata! {}
```

Comment:
This will not produce metadata. `metadata!` should be wrapping all impl section with contract methods. We should either remove it or wrap everything.

Reply:
Oh. Might be a leftover from `borsh`